packages/core/src/common/selection-command-handler.ts

```diff
@@ -28,17 +28,25 @@ export class SelectionCommandHandler<S> implements CommandHandler {
 
     execute(...args: any[]): Object | undefined {
         const selection = this.getSelection(...args);
-        return selection ? (this.options.execute as any)(selection, ...args) : undefined;
+        return selection ? (this.options.execute as any)(selection, ...args) : (this.options.execute as any)(undefined);
```

Comment:
That's a impactful change, we should analyze all clients of `SelectionCommandHandler` and make sure that they can handle missing args, an idea was that args are never missing. APIs for `this.options` should be adjusted to except no arguments.

Actually I don't like changing it. Instead we could subclass `SelectionCommandHandler` with `WorkspaceSelectionCommandHandler` and override `getSelection` which calls a super implementation and if it returns undefined or an empty list apply workspace root logic.